scripts/yosys/synth.tcl

```diff
@@ -335,6 +335,13 @@ proc run_strategy {output script strategy_name {postfix_with_strategy 0}} {
     tee -o "$::env(synth_report_prefix).$strategy_escaped.chk.rpt" check
     tee -o "$::env(synth_report_prefix).$strategy_escaped.stat.rpt" stat -top $::env(DESIGN_NAME) -liberty [lindex $::env(LIB_SYNTH_NO_PG) 0]
 
+    if { [info exists ::env(SYNTH_AUTONAME)] && $::env(SYNTH_AUTONAME) } {
+        # Generate public names for the various nets, resulting in very long names that include
+        # the full heirarchy, which is preferable to the internal names that are simply
+        # sequential numbers such as `_000019_`.
```

Comment:
should we give an example of what the name looks like here too?

Reply:
Sure, I've added that to the comment. It does get pretty gnarly, and appears to crash magic.

Note that klayout also doesn't work, but that's because OpenLane is generating invalid arguments to klayout -- klayout works fine if you pass it the correct arguments.